### ISSUE TITLE
Driver btrfs failed to remove root filesystem 

### ISSUE THREAD

**Issue by akaey** [docker/docker#22547]
**Output of `docker version`:**
```
Client:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 18:40:36 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.11.0
 API version:  1.23
 Go version:   go1.5.4
 Git commit:   4dc5990
 Built:        Wed Apr 13 18:40:36 2016
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**
```
Containers: 7
 Running: 6
 Paused: 0
 Stopped: 1
Images: 9
Server Version: 1.11.0
Storage Driver: btrfs
 Build Version: Btrfs v3.19.1
 Library Version: 101
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge null host
Kernel Version: 3.10.0-327.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 62.76 GiB
Name: cmx-docker1
ID: UHRB:KJT2:6QQS:ZDB4:NNJF:I47X:DGP6:6FYA:RFZX:CIHO:N3JN:L7QK
Docker Root Dir: /var/lib/docker
Debug mode (client): false
Debug mode (server): false
Http Proxy: http://proxy.esl.cisco.com:80/
Https Proxy: http://proxy.esl.cisco.com:80/
No Proxy: localhost,127.0.0.0/8,cmx-registry.cisco.com
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

VMWare ESXi VM with CentOS7

**Steps to reproduce the issue:**
1. Use the following docker-compose file and run "docker-compose up -d"

```yaml
haproxy:
  image: "haproxy:latest"
  container_name: dev-haproxy
  restart: always
  ports:
    - 80:80
    - 443:443
  volumes:
    - ./haproxy/haproxy.cfg:/usr/local/etc/haproxy/haproxy.cfg:ro
```

2. Since the haproxy.cfg file will be missing, it will go into restarting state. 
3. docker stop dev-haproxy
4. docker rm dev-haproxy

**Describe the results you received:**
```
Error response from daemon: Driver btrfs failed to remove root filesystem fc5ddc7eea06bf6a7cbadb42a3264f1901df674275f7b8598730b8d94332be19: Failed to destroy btrfs snapshot /var/lib/docker/btrfs/subvolumes for 60c3a4a226cb250df19d09d68677971369fb917c81b623ae0c2b38d4c7407a4d: device or resource busy
```
To recover from this, I need to stop the docker service and remove the subvolume with btrfs command

This is reproducible with other services like redis and cassandra if the config file is not there. When the container status is restarting, we cannot remove the btrfs subvolume. 

**Describe the results you expected:**
Expected result is that the btrfs volume should be deleted when removing the container. It should not show as busy. 

**Additional information you deem important (e.g. issue happens only occasionally):**
Always reproducible with bad docker-compose pointing to non-existent config file.

**Comment by thaJeztah** [comment]
Were you also having this issue in 1.10.3, or is this newly introduced in 1.11? Note that we released a 1.11.1 patch release; perhaps you can check if you are still able to reproduce on that version?

**Comment by akaey** [comment]
This is the first time setting up with btrfs so not sure about 1.10.3.

I upgraded to 1.11.1 patch and that seems to have solved the problem. When container status is restarting, I can stop the container and remove it without any btrfs errors. 

Thanks a lot.
